src/test/functional/functional-tests.js

```diff
@@ -55,16 +55,20 @@ try {
   minio = require('minio')
 }
 
-const { getVersionId, isArray,
-        CopyDestinationOptions,
-        CopySourceOptions,
-        removeDirAndFiles } = helpers
+const {
+  getVersionId, isArray,
+  CopyDestinationOptions,
+  CopySourceOptions,
+  removeDirAndFiles,
+  DEFAULT_REGION // comment/un comment for any testing.
+} = helpers
 
 require('source-map-support').install()
 
 const isWindowsPlatform = process.platform === "win32"
+const TEST_SERVER_REGION = DEFAULT_REGION  // use it for local testing.
 
-describe('functional tests', function() {
+describe('functional tests', function () {
```

Comment:
@prakashsvmx,
This comment is not about your change and it is an existing bug. It causes some unexpected behavior in 2 scenarios:
1- When the test runs against `play`. Check the lines  numbers 83 & 84, which set the accessKey and secretKey to the environment vars if env vars exist, even when we already know the endpoint is `play`. Then we know `play`'s accessKey and secretKey, so we should not check env vars at this point.
The code is supposed to be something like this:
```
  if (process.env['SERVER_ENDPOINT']) {
    var res = process.env['SERVER_ENDPOINT'].split(":")
    playConfig.endPoint = res[0]
    playConfig.port = parseInt(res[1])
    // Get the credentials from env vars, error out if they don't exist
    if (process.env['ACCESS_KEY]) {
        playConfig.accessKey = process.env['ACCESS_KEY']
    } else {
        // Error ACCESS_KEY had to be provided
    }
    if (process.env['SECRET_KEY']) {
        playConfig.secretKey = process.env['SECRET_KEY']
    } else {
        // Error SECRET_KEY had to be provided
    }
    // Set other config parameters and defaults
        :
  } else {
    playConfig.endPoint = 'play.min.io'
    playConfig.port = 9000
    playConfig.accessKey = 'Q3AM3UQ867SPQQA43P2F'
    playConfig.secretKey = 'zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG'
    // Set other config parameters and defaults
        :
  }
```

2- Running functinoal tests against a local server is broken. Here are the failure messages:
```

  functional tests
    1) "before all" hook in "functional tests"
    2) "after all" hook in "functional tests"


  0 passing (52ms)
  2 failing

  1) functional tests
       "before all" hook in "functional tests":
     Error: self signed certificate
      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
      at TLSSocket.emit (node:events:394:28)
      at TLSSocket._finishInit (node:_tls_wrap:944:8)
      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12)

  2) functional tests
       "after all" hook in "functional tests":
     Error: self signed certificate
      at TLSSocket.onConnectSecure (node:_tls_wrap:1530:34)
      at TLSSocket.emit (node:events:394:28)
      at TLSSocket._finishInit (node:_tls_wrap:944:8)
      at TLSWrap.ssl.onhandshakedone (node:_tls_wrap:725:12)



[00:04:45] '<anonymous>' errored after 750 ms
[00:04:45] Error in plugin "gulp-mocha"
Message:
    There were test failures
Details:
    domainEmitter: [object Object]
    domainThrown: false

[00:04:45] 'functionalTest' errored after 3.33 s
```

Reply:
Thank you. :+1:   I will investigate and send a fix.

Reply:
I have refactored the ENV settings block to make it clean.